nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -404,6 +404,87 @@ private Type.ClassType typeWithPreservedAnnotations(ParameterizedTypeTree tree)
     return finalType;
   }
 
+  // here we are checking if we should report an error on not considering the @Nullable annotations
+  // for overriden and
+  // overriding method return type should be consistent with the type parameter
+  public boolean DoNullabilityAnnotationsMatch(
```

Comment:
This should be split into two separate things: (1) updating the overriddenMethodType for type variables, and (2) checking that type parameter nullability matches.  For (1) we should just update the `overriddenMethodReturnNullness` at the caller and re-use the logic there for checking if there is a mismatch with the overriding method.  (2) should be in a separate method, or maybe we can even just expose a lightweight wrapper around `compareNullabilityAnnotations`